includes/platform-checkout/class-platform-checkout-order-status-sync.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Class Platform_Checkout_Webhooks
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Platform_Checkout;
+
+use WC_Payments_API_Client;
+use WCPay\Exceptions\API_Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class introduces webhooks to delivery order updates to the associated
+ * orders in the platform checkout.
+ *
+ * Platform Checkout Webhooks are enqueued to their associated actions, delivered, and logged.
+ */
+class Platform_Checkout_Order_Status_Sync {
+
+	/**
+	 * Client for making requests to the WooCommerce Payments API
+	 *
+	 * @var WC_Payments_API_Client
+	 */
+	protected $payments_api_client;
+
+	/**
+	 * Setup webhook for the Platform Checkout Order Status Sync.
+	 *
+	 * @param WC_Payments_API_Client $payments_api_client - WooCommerce Payments API client.
+	 */
+	public function __construct( WC_Payments_API_Client $payments_api_client ) {
+
+		$this->payments_api_client = $payments_api_client;
+
+		add_filter( 'woocommerce_webhook_topic_hooks', [ __CLASS__, 'add_topics' ], 20, 2 );
+		add_filter( 'woocommerce_webhook_payload', [ __CLASS__, 'create_payload' ], 10, 4 );
+		add_filter( 'woocommerce_valid_webhook_resources', [ __CLASS__, 'add_resource' ], 10, 1 );
+		add_filter( 'woocommerce_valid_webhook_events', [ __CLASS__, 'add_event' ], 10, 1 );
+		add_action( 'woocommerce_order_status_changed', [ __CLASS__, 'send_webhook' ], 10, 3 );
+
+		add_action( 'admin_init', [ $this, 'maybe_create_platform_checkout_order_webhook' ], 10 );
+	}
+
+	/**
+	 * Return the webhook name.
+	 *
+	 * @return string
+	 */
+	private static function get_webhook_name() {
+		return __( 'WCPay platform checkout order status sync', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Return the webhook delivery URL.
+	 *
+	 * @return string
+	 */
+	private static function get_webhook_delivery_url() {
+		$platform_checkout_host = defined( 'PLATFORM_CHECKOUT_HOST' ) ? PLATFORM_CHECKOUT_HOST : 'http://host.docker.internal:8090';
+		return $platform_checkout_host . '/wp-json/platform-checkout/v1/merchant-notification';
```

Comment:
@rafaelzaleski I'm getting this error when I try to update the Webhook on the merchant's site

![image](https://user-images.githubusercontent.com/1025173/173691271-c48c037a-2440-4373-9e6d-186d4936c117.png)